assets/js/components/Dashboard/PaMessaging/NewPaMessagePage/NewPaMessagePage.tsx

```diff
@@ -45,6 +45,10 @@ const NewPaMessagePage = () => {
 
   const navigate = useNavigate();
 
+  useEffect(() => {
+    if (!audioPlaying) setAudioReviewed(false);
+  }, [visualText, phoneticText]);
```

Comment:
Suppose I start playing the audio preview, then entirely within the duration of the preview, edit the text (maybe fixing a typo). In this case the audio would still count as "reviewed", despite the fact that I haven't actually previewed the audio for the new version of the text. This seems like a gap in the logic — maybe we are not really modeling the states correctly (in other words there should be an "audio has been reviewed" flag which is separate from the display state of the review button?).

Reply:
Mary made the decision to stop audio from playing when either textbox changes. This should eliminate the edge case you mentioned where audio is marked as reviewed if the text changes while playing. Now, the `onEnded` callback will only fire if the whole audio file is played without interruption.